src/pipeline.js

```diff
@@ -13,13 +13,8 @@ const _ = require('lodash/fp');
 const Promise = require('bluebird');
 
 const nopLogger = {
-  error: () => {},
```

Comment:
why did you remove this?
we should ensure that `action.logger` always contains a valid logger

Reply:
The `nopLogger` is only used in the Pipeline itself – and the removed methods are never called.

Reply:
Brought the functions back, but make sure they still get coverage, even when not called.